js/components/Compass/index.js

```diff
@@ -0,0 +1,110 @@
+import React, { Component } from 'react';
+import {
+  Image,
+  Animated,
+  DeviceEventEmitter
+} from 'react-native';
+import ReactNativeHeading from 'react-native-heading';
+
+export default class Compass extends Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      defaultAzimut: props.azimut || 360,
+      rotation: new Animated.Value(0),
+      headingIsSupported: false,
+      angleFilter: props.degreesFilter || 2
+    }
+  }
+
+  setFilter(filter) {
+    this.state.angleFilter = parseInt(filter);
+    this.stopHeadingUpdates();
+    this.startHeadingUpdates();
+  }
+
+  startHeadingUpdates() {
+    ReactNativeHeading.start(this.state.angleFilter)
+    .then(didStart => {
+        this.setState({
+            headingIsSupported: didStart,
+        })
+    });
+
+    DeviceEventEmitter.addListener('headingUpdated', (data) => {
+        this.startAnimation(data.heading || data);
+    });
+  }
+
+  stopHeadingUpdates() {
+    ReactNativeHeading.stop();
+    DeviceEventEmitter.removeAllListeners('headingUpdated');
+  }
+
+  componentWillMount(){
+    this.startAnimation(0);
+    this.startHeadingUpdates();
+  }
+
+  componentWillUnmount() {
+    this.stopHeadingUpdates();
+  }
+
+  startAnimation(z) {
+    const offset = this.calculateOffset();
+    if (offset != 0) {
```

Comment:
Yes it wont :), also the checking of the latitude and longitude is happening in calculateOffset so is safe to skip that check.